File: library/src/main/java/com/sa90/onepreference/adapter/viewholder/HeaderViewHolder.java
package com.sa90.onepreference.adapter.viewholder;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.sa90.onepreference.R;

/**
 * Created by Saurabh Arora on 21/11/16.
 */

public class HeaderViewHolder extends RecyclerView.ViewHolder {

    public TextView summary, title;
    public ImageView icon;

    public HeaderViewHolder(View itemView) {
        super(itemView);

        title = (TextView) itemView.findViewById(R.id.tvTitle);
        summary = (TextView) itemView.findViewById(R.id.tvSummary);
        icon = (ImageView) itemView.findViewById(R.id.icon);
    }
}

File: library/src/main/java/com/sa90/onepreference/adapter/HeaderViewHolder.kt
package com.sa90.onepreference.adapter

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.sa90.onepreference.R
import com.sa90.onepreference.model.Header

class HeaderViewHolder(itemView: View, itemClick: (Header) -> Unit) :
    RecyclerView.ViewHolder(itemView) {

    private val summary: TextView = itemView.findViewById(R.id.tvSummary)
    private val title: TextView = itemView.findViewById(R.id.tvTitle)
    private val icon: ImageView = itemView.findViewById(R.id.icon)

    init {
        itemView.setOnClickListener { itemClick(it.tag as Header) }
    }

    fun bind(header: Header) {
        itemView.tag = header
        if (header.iconRes == 0) {
            icon.visibility = View.GONE
        } else {
            icon.visibility = View.VISIBLE
            icon.setImageResource(header.iconRes)
        }
        title.text = header.getTitle(title.context.resources)
        val headerSummary = header.getSummary(title.context.resources)
        if (!headerSummary.isNullOrEmpty()) {
            summary.visibility = View.VISIBLE
            summary.text = headerSummary
        } else {
            summary.visibility = View.GONE
        }
    }

}